kv/kvserver/rangefeed: TestBudgetReleaseOnOneStreamError failed

kv/kvserver/rangefeed.TestBudgetReleaseOnOneStreamError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11273951?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11273951?buildTab=artifacts#/) on master @ [8ca9b057aa9cd6c0a91f52eb1389756b1e80f0f9](https://github.com/cockroachdb/cockroach/commits/8ca9b057aa9cd6c0a91f52eb1389756b1e80f0f9):


```
=== RUN   TestBudgetReleaseOnOneStreamError
    processor_test.go:1352: stream failed with error before all data was consumed retry rangefeed (REASON_SLOW_CONSUMER)
--- FAIL: TestBudgetReleaseOnOneStreamError (0.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBudgetReleaseOnOneStreamError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30517

--- a/pkg/kv/kvserver/rangefeed/processor_test.go
+++ b/pkg/kv/kvserver/rangefeed/processor_test.go
@@ -1301,7 +1301,7 @@ func TestBudgetReleaseOnOneStreamError(t *testing.T) {
 		PushTxnsInterval: pushTxnInterval,
 		PushTxnsAge:      pushTxnAge,
 		EventChanCap:     channelCapacity,
-		EventChanTimeout: time.Millisecond,
+		EventChanTimeout: 100 * time.Millisecond, // enough time for registration to process events
 		MemBudget:        fb,
 		Metrics:          NewMetrics(),
 	})

